app/models/reservation.rb

```
@@ -1,4 +1,5 @@
 class Reservation < ApplicationRecord
   belongs_to :user
   belongs_to :trade
+  validates :user_id, uniqueness: { scope: %i[trade_id date] }
 end
```

we could validate that the date is of reservation is not passed eg. reservation date 2023-09-30, todays date 2023-10-01. this should throw an error.